File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> count;
        for(int num : nums){
            count[num]++;
        }
        vector<pair<int,int>> arr;
        for(const auto& [num,occ] : count){
            arr.push_back({occ,num});
        }
        sort(arr.rbegin(),arr.rend());
        vector<int> result;
        for(int i = 0;i<k;i++){
            result.push_back(arr[i].second);
        }
        return result;
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        n = len(nums)
        l = 0
        r = n - 1
        while l < r:
            m = (l+r)//2
            if nums[m] > nums[r]:
                l = m + 1
            else:
                r=m
        d = l
        r = n - 1
        if target >= nums[d] and target <=nums[r]:
            l = d
            r = n - 1
        else:
            l = 0
            r = d
        while l <=r:
            m = (l+r)//2
            if nums[m] == target:
                return m
            elif nums[m] > target:
                r  = m - 1
            else:
                l = m + 1
        return -1